internal/pkg/service/buffer/worker/upload/reader_test.go

```diff
@@ -0,0 +1,67 @@
+package upload
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/c2h5oh/datasize"
+	gonanoid "github.com/matoous/go-nanoid/v2"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/schema"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+)
+
+func TestRecordsReader(t *testing.T) {
+	t.Parallel()
+
```

Comment:
Nevedel som, co sa stane, ked budeme mat v DB 5x1MB record, teda spolu `5MB`.
Page size mame by default v `iterator` nastavenu na `100`, teda tychto `5MB` by malo prist spolu.

V dokumentacii je `request` (nie `response`) size obmedzena na `1MB`, tak som si nebol isty, ci to nebude problem aj pri velkej response.

Ale vyzera to byt OK, test presiel aj s `1000` records, teda `10` pages.
![image](https://user-images.githubusercontent.com/19371734/213631576-a63812cb-c55c-4d4a-9bb5-d6c857262ee4.png)

Este som skusil aj vymenit `iterator` za `watcher` a nechat si `1000` records streamovat, ale to bolo pomalsie:
![image](https://user-images.githubusercontent.com/19371734/213631639-fe946da7-9dc9-4ca2-a57c-48546e8a7ba3.png)

(975MB, pretoze predtym som tam mal `999KB`, ... ono ten request limit je v skutocnosti `2MB` raw dat, nieco tam pridava `grpc` protokol, preto sa odporuca neprekracovat `1MB`).

Reply:
Test tu teda nechavam, ale problem tam nebol.